src/wymeditor/editor/dialogs.js

```diff
@@ -0,0 +1,501 @@
+        },
+        {
+            placeholder: WYMeditor.DIALOG_BODY,
+            replacement: pBodyHtml || dialog.getBodyHtml.call(wym)
+        }
+    ];
+
+    // Perform HTML template replacements
+    for (i = 0; i < htmlStrReplacements.length; i++) {
+        dialogHtml = WYMeditor.Helper.replaceAllInStr(
+            dialogHtml,
+            htmlStrReplacements[i].placeholder,
+            htmlStrReplacements[i].replacement
+        );
+    }
+
+    dialogHtml = wym.replaceStrings(dialogHtml);
+
+    doc = wDialog.document;
+    doc.write(dialogHtml);
+    if (dialog && dialog.getBodyClass) {
+        jQuery(doc.body).addClass(dialog.getBodyClass.call(wym));
+    }
+    doc.close();
+
+    selectedContainer = wym.selectedContainer();
+
+    wDialog.onbeforeunload = function () {
+        wym.focusOnDocument();
+    };
+
+    // pre-init functions
+    if (jQuery.isFunction(options.preInitDialog)) {
+        options.preInitDialog(wym, wDialog);
+    }
+
+    // auto populate fields if selected container (e.g. A)
+    if (selectedContainer) {
+        jQuery(options.hrefSelector, doc).val(jQuery(selectedContainer)
+            .attr(WYMeditor.HREF));
+        jQuery(options.srcSelector, doc).val(jQuery(selectedContainer)
+            .attr(WYMeditor.SRC));
+        jQuery(options.titleSelector, doc).val(jQuery(selectedContainer)
+            .attr(WYMeditor.TITLE));
+        jQuery(options.relSelector, doc).val(jQuery(selectedContainer)
+            .attr(WYMeditor.REL));
+        jQuery(options.altSelector, doc).val(jQuery(selectedContainer)
+            .attr(WYMeditor.ALT));
+    }
+
+    if (dialog && dialog.submitHandler) {
+        jQuery("form", doc).submit(function () {
+            dialog.submitHandler.call(wym, wDialog);
+        });
+    }
+
+    //cancel button
+    jQuery(options.cancelSelector, doc).click(function () {
+        wDialog.close();
+    });
+
+    //pre-init functions
+    if (jQuery.isFunction(options.postInitDialog)) {
+        options.postInitDialog(wym, wDialog);
+    }
+
+    return wDialog;
+};
+
+/*
+ * An object of default dialogs.
+ *
+ * Each has the following properties:
+ * String `title`
+ *     Dialog window title.
+ * Function `shouldOpen`
+ *     Its return value determines whether the dialog should be opened or not.
+ *     Is called with the editor as `this`.
+ * Function `getBodyHtml`
+ *     Used to provide the dialog's body's HTML. Is called with the editor as
+ *     `this`.
+ * Function `getBodyClass`
+ *     Optional. Returns a class that will be added to the body of the dialog
+ *     window's document.
+ * Function `getWindowFeatures`
+ *     Optional. Used to provide the dialog's window features, for passing to
+ *     `window.open`. Is called with the editor as `this`.
+ * function `SubmitHandler`
+ *     Optional. Handles a submit button press in the dialog. Is called with
+ *     the editor instance as `this`. Receives a single argument-the dialog
+ *     window.
+ */
```

Comment:
I love this! Is there any chance I could convince you to add a short bit of documentation under plugin authoring showing how to add a dialog that conforms to this? Once this makes it to a release, I'll rewrite PolicyStat's wymeditor plugins to use this.